test/metabase/lib/remove_replace_test.cljc

```diff
@@ -388,3 +389,114 @@
                   ;; TODO Should be able to create a ref with lib/field [#29763]
                   (lib/filter (lib/= [:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "a"] 1))
                   (lib/replace-clause 0 expr-a (lib/field "VENUES" "PRICE"))))))))
+
+(deftest ^:parallel replace-order-by-breakout-col-test
+  (testing "issue #30980"
+    (testing "Bucketing should keep order-by in sync"
+      (let [query (lib/query-for-table-name meta/metadata-provider "USERS")
+            breakout-col (->> (lib/breakoutable-columns query)
+                              (m/find-first (comp #{"LAST_LOGIN"} :name)))
+            month (->> (lib/available-temporal-buckets query breakout-col)
+                       (m/find-first (comp #{:month} :unit))
+                       (lib/with-temporal-bucket breakout-col))
+            day (->> (lib/available-temporal-buckets query breakout-col)
+                     (m/find-first (comp #{:day} :unit))
+                     (lib/with-temporal-bucket breakout-col))
+            q2 (-> query
+                   (lib/breakout month))
+            cols (lib/orderable-columns q2)
+            q3 (-> q2
+                   (lib/order-by (first cols))
+                   (lib/replace-clause (first (lib/breakouts q2)) day))
+            q4 (lib/replace-clause q3 (first (lib/breakouts q3)) month)]
+        (is (= :day (:temporal-unit (second (last (first (lib/order-bys q3)))))))
+        (is (= :month (:temporal-unit (second (last (first (lib/order-bys q4)))))))))
+    (testing "Binning should keep in order-by in sync"
```

Comment:
It's a little hard to read big 110 line tests like this and even harder to debug them when you break something, you could have broken this out into 5 separate `deftest`s since they're basically completely separate. I think splitting big tests out into a series of smaller ones is preferrable when it's easy to do, but since the 5 top-level `testing`s don't share anything it would have been relatively easy